src/components/Scheduler/Modals/PopupCreateContent.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { ISelectedRange } from '..';
+import { DateInput, Label, ModalButtonsWrapper, ModalContainer, ModalContent, ModalOverlay, SelectedDateText, Title, WrapperLabelAndInput } from './styles';
+import { CancelButton, ErrorText, SaveButton } from '../styles';
+
+interface IPopupCreateContentProps {
+    selectedDate: Date | undefined;
+    selectedRange: ISelectedRange;
+    errorMessage: string;
+    handleStartChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    handleEndChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    handleSave: () => void;
+    setShowCreatePopup: (show: boolean) => void;
+    setErrorMessage: (message: string) => void;
+};
+
+const PopupCreateContent = ({
+    selectedDate,
+    selectedRange,
+    errorMessage,
+    handleStartChange,
+    handleEndChange,
+    handleSave,
+    setShowCreatePopup,
+    setErrorMessage
+}: IPopupCreateContentProps) => {
+    const { t } = useTranslation();
+
+    return (
+        <ModalOverlay>
+            <ModalContainer>
+                <ModalContent>
+                    <Title>{t('Calendar.selectTime')}</Title>
+                    {errorMessage &&
+                    <ErrorText>{errorMessage}</ErrorText>}
+                    {selectedDate &&
+                    <SelectedDateText>{t('Calendar.selectedDate')} {selectedDate.toLocaleDateString()}</SelectedDateText>}
+                    <WrapperLabelAndInput>
+                        <Label htmlFor='start-time'>{t('Calendar.startTime')}</Label>
+                        <DateInput
+                            id='start-time'
+                            type='time'
+                            value={selectedRange?.start ?? ''}
+                            onChange={handleStartChange}
+                        />
+                    </WrapperLabelAndInput>
+                    <WrapperLabelAndInput>
+                    <Label htmlFor='end-time'>{t('Calendar.endTime')}</Label>
+                        <DateInput
+                            id='end-time'
+                            type='time'
+                            value={selectedRange?.end ?? ''}
+                            onChange={handleEndChange}
+                        />
+                    </WrapperLabelAndInput>
+                    <ModalButtonsWrapper>
+                        <CancelButton
+                            onClick={() => {
+                                setShowCreatePopup(false);
+                                setErrorMessage('');
+                            }}
```

Comment:
why its in render, not in separate function?

Reply:
moved to separate function